src/zql/query/entity-query.ts

```diff
@@ -163,6 +143,168 @@ export class EntityQueryImpl<S extends EntitySchema, Return = []>
   }
 
   prepare(): Statement<Return> {
-    return new Statement<Return>(this.#context, this);
+    return new Statement<Return>(this.#context, this.#ast);
   }
 }
+
+export class EntityGroupQuery<S extends EntitySchema, Return = []> {
+  readonly #ast: AST;
+  readonly #name: string;
+  readonly #context: Context;
+
+  constructor(context: Context, tableName: string, ast?: AST) {
+    this.#ast = ast ?? {
+      table: tableName,
+      alias: aliasCount++,
```

Comment:
What is the alias used for?

Reply:
it'll be used whenever we compile the query to SQL.

I think I mentioned this before but since the same table could be present many times (through joins and sub-queries) it'll be simplest to give each occurrence a unique alias.

Maybe it is irrelevant to do at this step and the AST -> SQL compiler can do this itself.